examples/todomvc/src/controller.rs

```diff
@@ -0,0 +1,124 @@
+//! The controller handles UI events, translates them into updates on the model,
+//! and schedules re-renders.
+
+use crate::todo::{Todo, TodoActions};
+use crate::todos::{Todos, TodosActions};
+use crate::visibility::Visibility;
+use dodrio::{RootRender, VdomWeak};
+use serde::{Deserialize, Serialize};
+
+/// The controller for the TodoMVC app.
+///
+/// This `Controller` struct is never actually instantiated. It is only used for
+/// its `*Actions` trait implementations, none of which take a `self` parameter.
+///
+/// One could imagine alternative controller implementations with `*Actions`
+/// trait implementations for (e.g.) testing that will assert various expected
+/// action methods are called after rendering TODO items and sending DOM events.
+#[derive(Default, Deserialize, Serialize)]
+pub struct Controller;
+
+/// Invoke `f` with the root `Todos` component, save the (likely just modified)
+/// todos to local storage, and schedule a new `dodrio` render.
+fn with_todos<F>(root: &mut dyn RootRender, vdom: VdomWeak, f: F)
```

Comment:
Using RAII may make this a bit more ergonomic to call by skipping some indentation maybe?

Reply:
Not more ergonomic to write initially though